framework/src/executioners/Transient.C

```diff
@@ -660,7 +663,7 @@ Transient::preExecute()
   */
 
   // Add time period start times to sync times
-  const std::vector<TimePeriod *> time_periods = _problem.getTimePeriods();
+  const std::vector<TimePeriodOld *> time_periods = _problem.getTimePeriods();
```

Comment:
There is tests and I believe it is used by some applications. This PR will deprecate it and I will remove it after I get all the warehouses updated.